design_docs/variable_mapping_reorg.md

```diff
@@ -0,0 +1,184 @@
+date: 2017/02/10 <br>
+</h1>
+<h2> Summary </h2>
+In discussions with @pwolfram, it became clear that we would like to keep
+mpas_xarray as general as possible, rather than adding code specific to
+MPAS-Analysis.  In particular, the capability for mapping variable names
+that is currently part of mpas_xarray is likely a capability that only
+MPAS-Analysis will need when opening xarray data sets.  Likewise, there is
+a desire for mpax_xarray not to use any of the functionality outside of its
+own module so that it remains autonomous from MPAS-Analysis.
+
+At the same time, it is desirable for efficiency and parallelism to perform
+certain operations during the preprocessing step within xarray, rather than
+constructing a data set first and then (in serial) performing manipulations
+(e.g. creating a time coordinate and slicing variables).
+
+The solution will be tested by making sure it produces bit-for-bit identical
+results to those from the develop branch for typical test cases on LANL IC
+and Edison.
+
+<h1> Requirements </h1>
+
+<h2> Requirement: mpas_xarray does not include MPAS-Analysis specific
+functionality <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+MPAS-Analysis specific functionality such as variable mapping should be 
+removed from mpas_xarray so it can remain an independent module, requiring
+minimal modification to accommodate MPAS-Analysis' needs.
+
+<h2> Requirement: MPAS-Analysis specific functionality should be supported in
+xarray preprossing <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+There should be a way to perform MPAS-Analysis specific functionality such as
+mapping variables during preprocessing.  This functionality should be
+relatively easy to add to as new preprocessing needs arise.
+
+
+<h1> Algorithmic Formulations (optional) </h1>
+
+<h2> Algorithm: mpas_xarray does not include MPAS-Analysis specific
+functionality <br>
+Date last modified: 2017/02/10 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+All functions and function arguments related to variable mapping will
+be removed from mpas_xarray and moved elsewhere.
+
+<h2> Algorithm: MPAS-Analysis specific functionality should be supported in
+xarray preprossing <br>
+Date last modified: 2017/02/15 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+A new utility function, `open_multifile_dataset` will added to `mpas_xarray` 
+that simplifies current calls to `xarray.open_mfdataset` to hide the 
+preprocessor and take care of removing redundant time indices once the dataset
+has been built. (This function doesn't directly address the requirement but
+is meant to make `mpas_xarray` easier to use and made sense because it
+has a one-to-one correspondence with other functionality, described below,
+that does address the requirement.)
+
+A new module, `generalized_reader` will also be added with its own
+`open_multifile_dataset` function.  This version takes additional arguments
```

Comment:
This is nice because it is clear that there is specific functionality being used for the opening operation.